test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/crypto/HapiCryptoUpdate.java

```diff
@@ -203,8 +207,14 @@ protected long feeFor(HapiApiSpec spec, Transaction txn, int numPayerKeys) throw
 						.setCurrentMemo(info.getMemo())
 						.setCurrentKey(info.getKey())
 						.setCurrentlyHasProxy(info.hasProxyAccountID())
+						.setCurrentMaxAutomaticAssociations(info.getMaxAutomaticTokenAssociations())
 						.build();
-				return cryptoOpsUsage.cryptoUpdateUsage(_txn, suFrom(svo), ctx);
+				var baseMeta = new BaseTransactionMeta(_txn.getMemoBytes().size(), 0);
+				var opMeta = new CryptoUpdateMeta(_txn.getCryptoUpdateAccount(),
+						_txn.getTransactionID().getTransactionValidStart().getSeconds());
+				var accumulator = new UsageAccumulator();
+				cryptoOpsUsage.cryptoUpdateUsage(suFrom(svo), baseMeta, opMeta, ctx, accumulator);
+				return AdapterUtils.feeDataFrom(accumulator);
```

Comment:
nit: it might be easier to undertand to move the code snippet into a separate function.